lms/djangoapps/labster_vouchers/views.py

```diff
@@ -0,0 +1,153 @@
+
+
+class ItemNotFoundError(Exception):
+    """
+    This exception is raised in the case where items is not found in Labster API.
+    """
+    pass
+
+
+class LabsterApiError(Exception):
+    """
+    This exception is raised in the case where problems with Labster API appear.
+    """
+    pass
+
+
+@require_http_methods(["GET"])
+@cache_control(no_cache=True, no_store=True, must_revalidate=True)
+@login_required
+def enter_voucher(request):
+    """
+    Enter Voucher View.
+    """
+    context = {'user': request.user}
+    return render_to_response('labster/enter_voucher.html', context)
+
+
+@require_http_methods(["POST"])
+@ensure_csrf_cookie
+@login_required
+def activate_voucher(request):
+    """
+    Gets license code from API, fetches related course_id and enrolls student.
+    """
+    form = forms.ValidationForm(request.POST)
+    enter_voucher_url = reverse('enter_voucher')
+
+    if not form.is_valid():
+        messages.error(request, "Enter a valid voucher code.")
+        return redirect(enter_voucher_url)
+
+    code = form.cleaned_data['code']
+
+    try:
+        license_code = get_license(code)
+    except ItemNotFoundError:
+        messages.error(request, _(
+            "Cannot find a voucher '{}'. Please contact Labster support team."
+        ).format(code))
+        return redirect(enter_voucher_url)
+    except LabsterApiError:
+        messages.error(request, _(
+            "There are some issues with applying your voucher. Please try again in a few minutes."
+        ))
+        return redirect(enter_voucher_url)
+
+    course_license = CourseLicense.objects.filter(license_code=license_code).first()
+
+    if not course_license:
+        messages.error(
+            request,
+            _("Cannot find a course for the voucher '{}'. Please contact Labster support team.").format(code)
+        )
+        return redirect(enter_voucher_url)
+
+    course_id = course_license.course_id
+    try:
+        # enroll student to course
+        add_enrollment(request.user, unicode(course_id))
+    except CourseNotFoundError:
+        messages.error(
+            request,
+            _(u"No course '{course_id}' found for enrollment").format(course_id=course_id)
+        )
+        return redirect(enter_voucher_url)
+    except CourseEnrollmentExistsError:
+        messages.error(
+            request,
+            _(u"You have been already enrolled to the course '{course_id}' before.").format(course_id=course_id)
+        )
+        return redirect(enter_voucher_url)
+    except CourseEnrollmentError:
+        messages.error(
+            request,
+            _(
+                u"An error occurred while creating the new course enrollment for user "
+                u"'{username}' in course '{course_id}'"
+            ).format(username=request.user.username, course_id=course_id)
+        )
+        return redirect(enter_voucher_url)
+
+    anon_uid = anonymous_id_for_user(request.user, course_id)
```

Comment:
> Does this mean that we create a new user for each course?

this function just returns user's anonymous id that is used by edX to communicate with 3rd party systems.